model/models.js

```diff
@@ -50,4 +50,52 @@ const fetchReviewById = (reviewId) =>
         return Promise.reject({status: 400, message: 'Invalid input!'});
     
 }
-module.exports = { fetchAllCategories, fetchAllReviews, fetchReviewById };
\ No newline at end of file
+
+const addComment = (reviewId, body) =>
+{
+    if(JSON.stringify(body) === '{}')
+        return Promise.reject({status: 400, message: 'request body is empty!'});
+    else
+    {
+        if(body.hasOwnProperty('username') && body.hasOwnProperty('body'))
+        {
+            const author = body.username;
+            const commentBody = body.body;
+            
+            return db.query(`SELECT review_id FROM reviews WHERE review_id = $1;`,[reviewId])
+            .then(({ rowCount }) => 
+            {
+                if(rowCount === 0)
+                    return Promise.reject({status: 404, message: 'review_id not found!'});
+                else
+                {
+                    return db.query(`SELECT username FROM users WHERE username = $1;`,[author])
+                    .then(({ rowCount }) =>
```

Comment:
If the username does not exist in the database, postGres will throw an error. This is not something that we need to check for as it's handled for us.

Reply:
I've removed this test case, thanks!